tests/test_grammar.py

```diff
@@ -255,8 +260,10 @@
     ("str_quoted_inter", "'${null}'", "None"),
     ("str_quoted_esc_single_1", r"'ab\'cd\'\'${str}'", "ab'cd''hi"),
     ("str_quoted_esc_single_2", r"""'\\\${foo}'""", r"\${foo}"),
+    ("str_quoted_esc_single_3", r"""'\\a_${str}\\'""", r" \\a_hi\ ".strip()),
```

Comment:
This is testing:
1. that double backslash is not unescaped before string interpolation
2. that double backslash is unescaped before a trailing quote.

I would prefer to split this into two different tests.

Reply:
Point 2 is actually tested already in the tests for quoted strings without interpolations:
```python
("str_quoted_trailing_esc_1", r"'abc\\'", r" abc\ ".strip()),
```
but I wanted to also have it tested with interpolations.
Anyway, I split it in two in f94e3da